feature_extractor/map_helper.py

```diff
@@ -89,110 +94,163 @@ def update(self, obs, game_state: GameState, entity_helper: EntityHelper):
       img = np.stack([tile_img, entity_img, poison_img, fog_img, visit_img, *coord_imgs])
       imgs.append(img)
-    imgs = np.stack(imgs)
-    return imgs
-
-  def nearby_features(self, row: int, col: int):
-    near_tile_map = self.tile_map[row-4:row+5, col-4:col+5]
-    food_arr = water_arr = herb_arr = fish_arr = obstacle_arr = []
-    for i in range(9):
-      for j in range(9):
-        if abs(i-4) + abs(j-4) <= 4:
-          food_arr.append(near_tile_map[i, j] == material.Forest.index)
-          water_arr.append(near_tile_map[i, j] == material.Water.index)
-          herb_arr.append(near_tile_map[i, j] == material.Herb.index)
-          fish_arr.append(near_tile_map[i, j] == material.Fish.index)
-          obstacle_arr.append(near_tile_map[i, j] in material.Impassible.indices)
-    food_arr[-1] = max(0, self.poison_map[row, col]) / 20.  # patch after getting trained
-    water_arr[-1] = max(0, self.poison_map[row+1, col]) / 20.  # patch after getting trained
-    herb_arr[-1] = max(0, self.poison_map[row, col+1]) / 20.  # patch after getting trained
-    fish_arr[-1] = max(0, self.poison_map[row-1, col]) / 20.  # patch after getting trained
-    obstacle_arr[-1] = max(0, self.poison_map[row, col-1]) / 20.  # patch after getting trained
-    # xcxc
-    return np.zeros(206)
-    return np.concatenate([
-        food_arr, water_arr, herb_arr, fish_arr, obstacle_arr,
-    ])
-
-  def legal_moves(self, obs):
+
+    return np.stack(imgs)
+
+  def nearby_features(self, row: int, col: int, nearby_dist=NEARBY_DIST):
+    # CHECK ME(kywch): my understanding of this function is to provide
+    #   especially important spatial features to the agent
+    #   such as nearby food, water, herb (poultice), fish (ration), obstacles, poison
+    #   As long as the order is fixed, agents will make sense of and use these features
+    near_tile_map = self.tile_map[row-nearby_dist//2:row+(nearby_dist//2+1),
+                                  col-nearby_dist//2:col+(nearby_dist//2+1)]
+    feat_arr = []
+    for i in range(nearby_dist):
+      for j in range(nearby_dist):
+        # (i,j) = (4,4) is the provided (row, col)
+        if abs(i-nearby_dist//2) + abs(j-nearby_dist//2) <= nearby_dist//2:
+          feat_arr.append(near_tile_map[i, j] == material.Forest.index) # food_arr
+          feat_arr.append(near_tile_map[i, j] == material.Water.index) # water_arr
+          feat_arr.append(near_tile_map[i, j] == material.Herb.index) # herb_arr
+          feat_arr.append(near_tile_map[i, j] == material.Fish.index) # fish_arr
+          feat_arr.append(near_tile_map[i, j] in material.Impassible.indices) # obstacle_arr
+
+        # adding poison map, which hadh comment: "patch after getting trained"
+        # CHECK ME: the below was set to <= 0, to make the output len 206
+        #   having wider poison map (like above) may help.
+        #   Changing it will require changing model.py, n_player_feat
```

Comment:
can we pull feature sizes into model_architecture.py, and use them both here and in model.py? that way we can dynamically scale both

Reply:
Would model/const.py be a good place?